pkgs/development/ruby-modules/gem/default.nix

```diff
@@ -179,7 +179,7 @@ stdenv.mkDerivation ((builtins.removeAttrs attrs ["source"]) // {
       '${version}' \
       '${lib.escapeShellArgs buildFlags}' \
       '${attrs.source.url}' \
-      '${src}' \
+      '.' \
```

Comment:
I think you want to use $NIX_BUILD_TOP here.

Reply:
Ah, didn't know about that variable. If I do `"$NIX_BUILD_TOP" \` for that line, I still get this error on my test case:

```terminal
$ nix-build
these derivations will be built:
  /nix/store/3hfcigzb57ijdvs6msn0ik4lzk73s9nv-ruby2.5.5-http_parser.rb-0.6.1.drv
  /nix/store/ana9rq1r399wnmvlv9kxw76zxmjdqgqg-ruby-env.drv
building '/nix/store/3hfcigzb57ijdvs6msn0ik4lzk73s9nv-ruby2.5.5-http_parser.rb-0.6.1.drv'...
unpacking sources
unpacking source archive /nix/store/rfgsdvv1mxi6kj07xiijhzhz8dps65h3-http_parser.rb-225354b
source root is http_parser.rb-225354b
patching sources
configuring
no configure script, doing nothing
building
Initialized empty Git repository in /build/http_parser.rb-225354b/.git/
installing
buildFlags:
`/homeless-shelter` is not a directory.
Bundler will use `/build/bundler/home/unknown' as your home directory temporarily.
/nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:1227:in `stat': No such file or directory @ rb_file_s_stat - /build/.git (Errno::ENOENT)
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:1227:in `lstat'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:1256:in `copy'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:418:in `block in copy_entry'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:1390:in `wrap_traverse'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:415:in `copy_entry'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:393:in `block in cp_r'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:1463:in `block in fu_each_src_dest'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:1479:in `fu_each_src_dest0'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:1461:in `fu_each_src_dest'
        from /nix/store/rwp5fpzqssf5m9dzbgbwsfgdzw8xajra-ruby-2.5.5/lib/ruby/2.5.0/fileutils.rb:392:in `cp_r'
        from /nix/store/6g99037s460cbv2drh4l7hrny09lc19q-nix-bundle-install.rb:92:in `checkout'
        from /nix/store/q3m22q8z2rsq1hjy6lsv4amavyf872lw-bundler-1.17.2/lib/ruby/gems/2.5.0/gems/bundler-1.17.2/lib/bundler/source/git.rb:303:in `fetch'
        from /nix/store/q3m22q8z2rsq1hjy6lsv4amavyf872lw-bundler-1.17.2/lib/ruby/gems/2.5.0/gems/bundler-1.17.2/lib/bundler/source/git.rb:161:in `specs'
        from /nix/store/6g99037s460cbv2drh4l7hrny09lc19q-nix-bundle-install.rb:119:in `<main>'
builder for '/nix/store/3hfcigzb57ijdvs6msn0ik4lzk73s9nv-ruby2.5.5-http_parser.rb-0.6.1.drv' failed with exit code 1
cannot build derivation '/nix/store/ana9rq1r399wnmvlv9kxw76zxmjdqgqg-ruby-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/ana9rq1r399wnmvlv9kxw76zxmjdqgqg-ruby-env.drv' failed
```

Reply:
I suppose there's not. It seems weird to have the parameter that only receives (AFAICT) a constant value, though. Either it should be removed as a parameter (and the receiving script updated appropriately) or the variation is significant and there's something we're missing?

Thanks for making the experiment. I'm sorry for the bum steer.

Reply:
Yeah, I'm quite curious about wether I missed something. Makes sense what you say about that constant parameter. Let's see wether someone else finds a problem here before I refactor that.